libp2p/crypto/ecnist.nim

```diff
@@ -396,20 +392,12 @@ proc toBytes*(seckey: EcPrivateKey, data: var openArray[byte]): EcResult[int] =
     p.finish()
     b.write(p)
     b.finish()
-    var blen = len(b)
-    if len(data) >= blen:
-      copyMem(addr data[0], addr b.buffer[0], blen)
-    # ok anyway, since it might have been a query...
-    ok(blen)
+    ok(b.buffer[0 ..< b.len])
```

Comment:
`b.buffer[0 ..< b.len]` will create new copy of bytes, why is it needed? 
and other instances...